src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -2049,6 +2051,57 @@ export class CompareWithSavedAction extends Action {
 	}
 }
 
+export class CompareWithClipboardAction extends Action {
+
+	public static ID = 'workbench.files.action.compareWithClipboard';
+	public static LABEL = nls.localize('compareWithClipboard', "Compare Active File with Clipboard");
+
+	private static SCHEME = 'compareWithClipboard';
+
+	private toDispose: IDisposable[];
+
+	constructor(
+		id: string,
+		label: string,
+		@IWorkbenchEditorService private editorService: IWorkbenchEditorService,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@ITextModelService textModelService: ITextModelService,
+	) {
+		super(id, label);
+
+		this.enabled = true;
+		this.toDispose = [];
+
+		const provider = instantiationService.createInstance(ClipboardContentProvider);
+		// this.toDispose.push(provider);
+
+		const registrationDisposal = textModelService.registerTextModelContentProvider(CompareWithClipboardAction.SCHEME, provider);
+		this.toDispose.push(registrationDisposal);
+	}
+
+	public run(): TPromise<any> {
+		let resource: URI = toResource(this.editorService.getActiveEditorInput(),
+			{ supportSideBySide: true, filter: 'file' }
+		);
+
+
+		if (resource && resource.scheme === 'file') {
```

Comment:
You do not need to check for `resource.scheme === 'file'` again, the previous call makes sure of that.